src/scene/materials/material.js

```diff
@@ -97,14 +97,20 @@ Object.assign(pc, function () {
         this._shaderVersion = 0;
         this._scene = null;
         this._dirtyBlend = false;
+
+        this.dirty = true;
     };
 
     Object.defineProperty(Material.prototype, 'shader', {
         get: function () {
             return this._shader;
         },
         set: function (shader) {
-            this.setShader(shader);
+            if (this._shader) {
+                this._shader._refCount--;
+            }
+            this._shader = shader;
+            if (shader) shader._refCount++;
```

Comment:
Is this refCount new?

Reply:
No. If you look at the whole PR, this code was cut and pasted from pc.Material#setShader, which now just sets the shader property and has been officially deprecated.